src/reactive/duplicity.py

```diff
@@ -56,6 +69,7 @@ def install_duplicity():
     fetch.apt_install("python-paramiko")
     fetch.apt_install("python-boto")
     fetch.apt_install("lftp")
+    install_in_system_python("azure-storage-blob")
```

Comment:
**nitpick:** 

Is the package has to be installed even it's not azure?
Maybe you want to move this step to `config-changed` event handler?

Also you should consider the issue if network is not available.


```python
ok = install_in_system_python("azure-storage-blob")
if ok:
    set_flag("duplicity.azure_package_installed")
else:
    # raise error so user can retry with `juju-resolve`
```

Reply:
hi @ashugauttam225 , the error handling is still missing here.
Do you want to have that in this PR or we just raise error and let charm get into error status?